test.py

```diff
@@ -458,6 +462,153 @@ def test_schedule_purge(self):
+        with open(join(path, 'foobar'), 'w') as f:
+            f.write('foobar')
+        # check we have no schedule
+        resp = self.app.get('/VolumeDriver.Schedule.List')
+        schedule = json.loads(resp.body.decode())['Schedule']
+        self.assertEqual(len(schedule), 0)
+        # check we have no snapshots
+        resp = self.app.post('/VolumeDriver.Snapshot.List', json.dumps({}))
+        snapshots = json.loads(resp.body.decode())['Snapshots']
+        self.assertEqual(len(snapshots), 0)
+        # synchronize the volume every 120 minutes, even some host are not
+        # responding we should synchronise other hosts
+        self.app.post('/VolumeDriver.Schedule', json.dumps(
+            {
+                'Name': name,
+                'Action': 'synchronize:localhost,wronghost.mlf',
+                'Timer': 120
+            }))
+        # also replicate a non existing volume
+        self.app.post('/VolumeDriver.Schedule', json.dumps(
+            {'Name': 'boo', 'Action': 'synchronize:localhost', 'Timer': 120}))
+        # simulate we spent more time
+        SCHEDULE_LOG.setdefault(
+            'synchronize:localhost,wronghost.mlf', {}
+        )
+        SCHEDULE_LOG['synchronize:localhost,wronghost.mlf'][name] = \
+            datetime.now() - timedelta(1)
+        with TemporaryDirectory(remote_path) as remote_path:
+            with open(join(remote_path, 'foobar'), 'w') as f:
+                f.write('test sync')
+            # run the scheduler and check we only have two more snapshots
+            scheduler(SCHEDULE, test=True)
+        # make sure a snapshot has occure before rsync
+        snapshots = [s for s in os.listdir(SNAPSHOTS_PATH)
+                     if s.startswith(name)]
+        self.assertEqual(
+            1, len(snapshots))
+
+        with open(join(SNAPSHOTS_PATH, snapshots[0], 'foobar')) as x:
+            self.assertEqual(x.read(), "foobar")
+        with open(join(path, 'foobar')) as x:
+            self.assertEqual(x.read(), "test sync")
+
+        # unschedule the last job
+        self.app.post('/VolumeDriver.Schedule', json.dumps(
+            {'Name': 'boo', 'Action': 'synchronize:localhost', 'Timer': 0}))
+        self.app.post('/VolumeDriver.Schedule', json.dumps(
+            {
+                'Name': name,
+                'Action': 'synchronize:localhost,wronghost.mlf',
+                'Timer': 0,
+            }
+        ))
+
+
+class TemporaryDirectory(object):
+    """Create and return a temporary directory. This has the same
+    behavior as tempfile.TemporaryDirectory but is used with a given path
+    directory, if directory already exists that directory and everything
+    contained in it are removed.  For
+    example:
+        with TemporaryDirectory('/tmp/mydir') as tmpdir:
+            ...
+    Upon exiting the context, the directory and everything contained
+    in it are removed.
+    """
+
+    def __init__(self, path):
+        self.name = self.mkdir(path)
```

Comment:
Can't you just inherit TemporaryDirectory ? It would save many lines and avoid copy-paste.
Something probably like:
```
class ExplicitTmpDir(TemporaryDirectory):
    def __init__(self, suffix=None, prefix=None, dir=None, path=None):
        self.name = path or mkdtemp(suffix, prefix, dir)
        self._finalizer = weakref.finalize(
              self, self._cleanup, self.name,
              warn_message="Implicitly cleaning up {!r}".format(self))
```